Makefile

```diff
@@ -50,9 +50,17 @@ book: ## Builds the book & serves documentation site
 
 # --- testing -------------------------------------------------------------------------------------
 
+.PHONY: test-build
+test-build: ## Build the test binary
+	cargo nextest run --cargo-profile test-release --features concurrent,testing --no-run
+
 .PHONY: test
-test: ## Runs all tests with the release profile
-	$(DEBUG_ASSERTIONS) cargo nextest run --cargo-profile test-release --features testing
+test: ## Run all tests
+	$(DEBUG_ASSERTIONS) cargo nextest run --profile default --cargo-profile test-release --features concurrent,testing
```

Comment:
I would remove the `$(DEBUG_ASSERTIONS)`. I believe this can cause a full rebuild when running `make test` even if you just completed a `make test-build`. I would remove it wherever the debug assertions are already specified in the cargo profile, so whenever you use `test-release`.

The same might be true for the `$(WARNINGS)` below.

Reply:
> The same might be true for the $(WARNINGS) below.

FYI the `$(WARNINGS)` is still there for `test-doc` in `miden-base`, not sure if that's intended then

Reply:
Ah thanks. Actually, since this is run in a separate job without a build step this is probably fine. Though could be worth looking into if that means it does not reuse the cache. In any case, test-docs wasn't the longest job overall so I didn't prioritize it.